### PR TITLE
Prescriptive guidance on restoring tables and databases

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-6903

This adds a couple of paragraphs to the restore and backup docs for further detail around what is/isn't backed up/restored from a table backup when there are object dependencies.

In a push to provide some more prescriptive guidance to the DR docs, paragraphs include recommendations on favoring database level backup where there are tables with object dependencies.

### CODE REVIEW

File: src/current/_includes/v23.1/backups/object-dependency.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+When you back up a table, the backup will include all of its dependent objects, but the backup will not include any dependent tables. For example, if you back up [view]({% link {{ page.version.version }}/views.md %}) `v` that depends on table `t`, it will only back up `v`, not `t`. When you try to restore `v`, the restore will fail because the referenced table is not present in the backup. Alternatively, you can pass [`skip_missing_views`]({% link {{ page.version.version }}/restore.md %}#skip-missing-views) with {% if page.name == "restore.md" %} `RESTORE` {% else %} [`RESTORE`]({% link {{ page.version.version }}/restore.md %}) {% endif %} to skip the view instead. This is also the case with tables or views with [sequences]({% link {{ page.version.version }}/create-sequence.md %}), you will receive an error in the same way, or you need to pass [`skip_missing_sequences`]({% link {{ page.version.version }}/restore.md %}#skip-missing-sequences) with `RESTORE`.
```
Comment (kathancox) [inline src/current/_includes/v23.1/backups/object-dependency.md:1]: Here's the rendered preview for this: https://deploy-preview-18225--cockroachdb-docs.netlify.app/docs/stable/restore#object-dependencies

File: src/current/_includes/v23.1/backups/object-dependency.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+When you back up a table, the backup will include all of its dependent objects, but the backup will not include any dependent tables. For example, if you back up [view]({% link {{ page.version.version }}/views.md %}) `v` that depends on table `t`, it will only back up `v`, not `t`. When you try to restore `v`, the restore will fail because the referenced table is not present in the backup. Alternatively, you can pass [`skip_missing_views`]({% link {{ page.version.version }}/restore.md %}#skip-missing-views) with {% if page.name == "restore.md" %} `RESTORE` {% else %} [`RESTORE`]({% link {{ page.version.version }}/restore.md %}) {% endif %} to skip the view instead. This is also the case for tables or views with [sequences]({% link {{ page.version.version }}/create-sequence.md %}); you can use the [`skip_missing_sequences`]({% link {{ page.version.version }}/restore.md %}#skip-missing-sequences) option with `RESTORE`.
```
Comment (kathancox) [inline src/current/_includes/v23.1/backups/object-dependency.md:1]: Here is the rendered preview for this: https://deploy-preview-18225--cockroachdb-docs.netlify.app/docs/stable/restore#object-dependencies

**Review by stevendanna — COMMENTED** [review]
I'm a bit torn here.  I appreciate that we want to provide more prescriptive guidance, but also want to make sure it is easily actionable.

For example 

> As you design your schema, if a database contains some tables that have dependent objects alongside independent tables, you will have greater flexibility by separating them into different databases.

Might not be easily possible for the user.

Overall, I wonder if we might take a different path. Namely, the section this is going already talks about dependent objects. It starts with:

> Dependent objects must be backed up at the same time as the objects they depend on.

Maybe we could expand that to be slightly more clear and then a follow up sentence with the example.  Then we have the table of the object you are backing up and the dependencies they require. Perhaps we should expand that with another column of the option that allows you to skip those dependencies.

Then after that table, perhaps we have some advice. I think that advice should probably be that we recommend database-level restores when possible.

**Comment by kathancox** [comment]
@stevendanna Got it, yeah "prescriptive guidance" was a push from PM/EM a couple of months back, but I think we can balance this a bit more as you say. I have framed this a bit differently based on your suggestions.

**Review by stevendanna — APPROVED** [review]
This seems like a good improvement to me.

One sentence I'm unsure on is:

> When you back up a table, the backup will include all of its dependent objects, but the backup will not include any dependent tables.

What's an object? This confusion is slightly made worse by the column heading "Object" below which then includes tables.  Unfortunately, I don't have a better word here. Perhaps object is the right word.

File: src/current/_includes/v23.1/backups/object-dependency.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+Dependent object | Depends on | Skip option
+-------|------------+-------------
+Table with [foreign key]({% link {{ page.version.version }}/foreign-key.md %}) constraints | The table it `REFERENCES`. | [`skip_missing_foreign_keys`]({% link {{ page.version.version }}/restore.md %}#skip_missing_foreign_keys)
+Table with a [sequence]({% link {{ page.version.version }}/create-sequence.md %}) | The sequence. | [`skip_missing_sequences`]({% link {{ page.version.version }}/restore.md %}#skip-missing-sequences)
+[Views]({% link {{ page.version.version }}/views.md %}) | The tables used in the view's `SELECT` statement. | [`skip_missing_views`]({% link {{ page.version.version }}/restore.md %}#skip-missing-views)
+
+We recommend treating tables with [foreign keys]({% link {{ page.version.version }}/foreign-key.md %}), which contribute to [views]({% link {{ page.version.version }}/views.md %}), or that use sequences or user-defined types as a single unit. While you can restore individual tables, you may find that backing up and restoring at the database level is more convenient.
```
Comment (florence-crl) [inline src/current/_includes/v23.1/backups/object-dependency.md:13]:
```suggestion
We recommend treating tables with [foreign keys]({% link {{ page.version.version }}/foreign-key.md %}), which contribute to [views]({% link {{ page.version.version }}/views.md %}), or that use sequences or user-defined types as a single unit with their dependencies. While you can restore individual tables, you may find that backing up and restoring at the database level is more convenient.
```

**Review by florence-crl — APPROVED** [review]
LGTM with 1 possible clarifying clause.

File: src/current/_includes/v23.1/backups/object-dependency.md (hunk 3 above)
Comment (kathancox) [reply src/current/_includes/v23.1/backups/object-dependency.md]: Done! Thanks!

**Pull request merged**